docs/python/debugging.md

```diff
@@ -55,11 +55,214 @@ Selecting the **Node.js: Gulp task** yields the following result:
+
+![Start debugging](images/debugging/debug-start-button.png)
+
+## Command line debugging
+
+The debugger can also be run from the command line. The debugger command line syntax is as follows:
+
+```bash
+python -m debugpy --listen [<address>:]<port>
+    [--wait-for-client]
+    [--configure-<name> <value>]...
+    [--log-to <path>] [--log-to-stderr]
+    <filename> | -m <module> | -c <code> | --pid <pid>
+    [<arg>]...
+```
+
+As an example, from the command line, you could start the debugger using a specified port (5678) and script using the following syntax. This example assumes the script is long running and omits the `--wait-for-client` flag, meaning that the script will not wait for the client to attach.
+
+```bash
+python -m debugpy --listen 5678 ./myscript.py
+```
+
+You would then use the following configuration to attach from the VS Code Python extension.
+
+```json
+{
+    "name": "Python: Attach",
+    "type": "python",
+    "request": "attach",
+    "host": "localhost",
+    "port": 5678,
+}
+```
+
+If you wanted to debug remote code or code running in a docker container, on the remote machine or container, you would need to modify the previous CLI command to specify a host.
+
+```bash
+python -m debugpy --listen 0.0.0.0:5678 ./myscript.py
+```
+
+The associated configuration file would then look as follows.
+
+```json
+{
+    "name": "Attach",
+    "type": "python",
+    "request": "attach",
+    "host": "remote-machine-name", // replace this with remote machine name
+    "port": 5678,
+}
+```
+
+> **Note**: Be aware that when you specify a host value other than `127.0.0.1` or `localhost` you are opening a port to allow access from any machine, which carries security risks. You should make sure that you're taking appropriate security precautions, such as using SSH tunnels, when doing remote debugging.
+
+|Flag  |Options  |Description  |
+|---------|---------|---------|
+|**--listen**     |  `[<address>:]<port>`       |   **Required**. Specifies the host address and port for the debug adapter server. This is the same address that is used in the VS Code debug configuration. By default the host address is `localhost (127.0.0.1)`.      |
+|**--wait-for-client**     |   none      | **Optional**. Specifies that the code should not run until there's a connection from the debug server. This setting allows you to debug from the first line of your code.        |
+|**--log-to**     |   `<path>`      | **Optional**. Specifies a path to an existing directory for saving logs.         |
+|**--log-to-stderr**     |    none     |  **Optional**. Enables debugpy to write logs directly to stderr.       |
+|**--pid**     |    `<pid>`     | **Optional**. Specifies a process that is already running that the debugger should attach to.        |
```

Comment:
When used directly from CLI, `--pid` simply injects the debug server into the process - no actual attach happens. Rather, it's as if (best effort) the targeted process were started with `--connect` or `--listen` as used along with `--pid`. It is used under the hood to implement "attach" with "processId", but that's an implementation detail that users don't need to be concerned about.

Reply:
Thanks, reworded for next commit.